### ISSUE TITLE
Support storage scopes for Android

### ISSUE THREAD

**Issue by k4r4b3y** [amake/orgro#123]
Newer android versions have storage scope permissions which allow the user to constrain the files and folders the app can read and write.  Orgro currently doesn't use the storage scopes permissions on android.  Supporting that would increase the user privacy and "sandbox" the Orgro app with regards to which files and directories it can reach.

**Comment by amake** [comment]
Orgro already does not request any permissions to access storage; all access is mediated through intents and the [Storage Access Framework](https://developer.android.com/guide/topics/providers/document-provider).

Can you please be more specific about what changes you are expecting to see in Orgro?

**Comment by k4r4b3y** [comment]
Take Orgzly Revived app for android for example.  When I long press on the app on Android, I have "App info", "Widgets", AND, "**Storage Scopes**" pop-up menu options.  When I tap on **Storage Scopes**, Android presents me with a UI with three buttons: Add folder, Add file, and Add image.  When I tap on Add folder, I can select which folder the Orgzly app is _allowed_ to use.  Any other directory is outside of the App's user-allowed Storage Scope, and is denied automatically.

I would like to see Orgro also using this Android privacy setting.

**Comment by amake** [comment]
I installed Orgzly Revived in an Android 15 emulator and I don't see the UI you describe. Can you provide a screenshot?

**Comment by amake** [comment]
I note that Orgzly Revived requests a bunch of storage-related permissions:
https://github.com/orgzly-revived/orgzly-android-revived/blob/b774a06852fdf8e4750dc6d5c201a2d3658d694d/app/src/main/AndroidManifest.xml#L12-L18

Also please note that Orgro requests *none* of those permissions (look carefully: I actually *remove* a unnecessary permission mistakenly requested by a plugin; see https://github.com/hpoul/file_picker_writable/issues/15):
https://github.com/amake/orgro/blob/d91e414e159d6d3c2d3f26f292806c4ea5f3d066/android/app/src/main/AndroidManifest.xml#L4-L6

Orgro is already *much* more constrained than Orgzly Revived. I suspect the only reason your OS is mentioning scopes for Orgzly is *because* it requests so many permissions.

In other words there is nothing to do here. Orgro is already locked down as tight as it can be.

**Comment by k4r4b3y** [comment]
> Can you provide a screenshot?

![storage-scope-01](https://github.com/user-attachments/assets/c07e32da-de9b-48a4-a6b2-6db241dc7950)
![storage-scope-02](https://github.com/user-attachments/assets/9932609b-12d9-44e4-a895-978862a2525d)

**Comment by amake** [comment]
Thanks. I don't have that section on my emulator, but...

I see in this discussion that functionally Storage Scopes are like a "jailed" full-disk access permission:
https://discuss.grapheneos.org/d/1935-how-do-i-enable-storage-scopes/2

I imagine I may be able to get that UI if I do something in Orgzly that would trigger it to ask for full-disk access, but I'm not familiar enough with the app to figure it out.

Regardless, Orgro never asks for such permission, nor does it need it. So this UI is simply not relevant to Orgro.

**Comment by k4r4b3y** [comment]
>Regardless, Orgro never asks for such permission, nor does it need it. So this UI is simply not relevant to Orgro.

Alright then.
